token-swap/program/fuzz/src/lib.rs

```diff
@@ -0,0 +1,586 @@
+//! Helpers for working with swaps in a fuzzing environment
+
+use spl_token_swap::{
+    curve::base::SwapCurve,
+    instruction::{self, Deposit, Swap, Withdraw},
+    state::SwapInfo,
+};
+
+use spl_token::{
+    instruction::approve,
+    state::{Account as TokenAccount, AccountState as TokenAccountState, Mint},
+};
+
+use solana_program::{
+    account_info::AccountInfo, bpf_loader, clock::Epoch, entrypoint::ProgramResult,
+    instruction::Instruction, program_error::ProgramError, program_option::COption,
+    program_pack::Pack, program_stubs, pubkey::Pubkey, system_program,
+};
+
+struct TestSyscallStubs {}
+impl program_stubs::SyscallStubs for TestSyscallStubs {
+    fn sol_invoke_signed(
+        &self,
+        instruction: &Instruction,
+        account_infos: &[AccountInfo],
+        signers_seeds: &[&[&[u8]]],
+    ) -> ProgramResult {
+        let mut new_account_infos = vec![];
+
+        // mimic check for token program in accounts
+        if !account_infos.iter().any(|x| *x.key == spl_token::id()) {
+            return Err(ProgramError::InvalidAccountData);
+        }
+
+        for meta in instruction.accounts.iter() {
+            for account_info in account_infos.iter() {
+                if meta.pubkey == *account_info.key {
+                    let mut new_account_info = account_info.clone();
+                    for seeds in signers_seeds.iter() {
+                        let signer =
+                            Pubkey::create_program_address(&seeds, &spl_token_swap::id()).unwrap();
+                        if *account_info.key == signer {
+                            new_account_info.is_signer = true;
+                        }
+                    }
+                    new_account_infos.push(new_account_info);
+                }
+            }
+        }
+
+        spl_token::processor::Processor::process(
+            &instruction.program_id,
+            &new_account_infos,
+            &instruction.data,
+        )
+    }
+}
```

Comment:
This is lifted directly from the processor test code, not sure how we can share this better.